### DISCUSSION TITLE
[4.x] 'Has a value'/'is empty' field condition for most field types

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#11070] [Ideas] [answered]
When adding an entry condition rule, the available options for relational fields include `has a value` and `is empty`.

I think it would make sense to also add these options to most other field types. So, for example, a _text field_ would have `equals`,`begins with`, `ends with`, `contains`, `has a value`, `is empty`.

**Comment by simonkuran** [comment]
I would use this all the time!

**Comment by ballmannweber** [comment]
I was surprised the conditional didn’t has this option because we would love to use it in terms of user guidance.

There is often the problem that customers insert a new entry because, for example, they have a new product or a new photo, a showroom or something similar. Then they create a new entry, but don't give it a title at first, but go to the file field and load this one image into it. Since they are in a hurry, they write something in the title, often XXX or some nonsense. 

The entry is then completed days or weeks later, until then it only has the function of a reminder in the system. However, since the slug is not automatically adjusted, you end up with a lot of incorrect slugs. This can also happen with other fields, depending on what content the customer wants to get rid of "quickly".

That's why I would like to use such a conditional for user guidance: First, only the (title) field appears, which has to be filled in before any other fields are displayed.

The topic of user guidance and backend design is a very important one - and the more the CMSs become similar in the backend, the more the focus will shift to customisability in the backend.

We now see the backend as almost more important than the frontend, because the quality and also the runtime of the website stands and falls with good or bad maintenance.

**Reply by simonkuran** [reply]
How would that encourage realistic titles? Couldn't the editor just put in a placeholder like they already do and move on to the other fields?

Maybe an option for the slugs only be generated when the entry goes live for the first time could help, but that's probably a can of worms for another feature request.

**Reply by ballmannweber** [reply]
Right, there is no guarantee that the user will fill in the title correctly, but it gives them an idea of the order in which to proceed.

**Comment by brandonkelly** [comment, marked as answer]
Added “has a value” and “is empty” operators to all text, number, and date range rules for Craft 4.3 (via #12135) 🎉

**Reply by brandonkelly** [reply]
Craft 4.3 was released earlier this week with this!
